lms/djangoapps/mobile_api/models.py

```diff
@@ -25,3 +26,63 @@ def get_video_profiles(cls):
         Get the list of profiles in priority order when requesting from VAL
         """
         return [profile.strip() for profile in cls.current().video_profiles.split(",") if profile]
+
+
+class MobileAppVersionConfig(ConfigurationModel):  # pylint: disable=model-missing-unicode
+    """
+    Configuration for mobile app versions available.
+    """
+    IOS = "ios"
+    ANDROID = "android"
+    MOBILE_PLATFORM = (
+        (IOS, "IOS"),
+        (ANDROID, "ANDROID"),
+
+    )
+    KEY_FIELDS = ('platform', )  # mobile platform is unique
+    platform = CharField(
+        max_length=50,
+        choices=MOBILE_PLATFORM,
+        blank=False,
+        help_text="mobile device platform")
+    latest_version = TextField(blank=False, help_text="Latest available version for app in X.X.X format")
+    min_supported_version = TextField(blank=False, help_text="Min supported version for app in X.X.X format")
+    next_min_supported_version = TextField(
+        blank=False,
+        help_text="Next supported version for app in X.X.X format that a user must upgrade to before deadline"
+    )
+    update_required_date = DateTimeField(verbose_name="Upgrade Deadline")
+
+    def parsed_version(self, version):
+        """ Converts string X.X.X.Y to tuple (X, X, X, Y) and return (X, X, X) """
+        return tuple(version.split("."))[:3]
+
+    def is_outdated_version(self, user_app_version):
+        """
+        This is the case if user app version is no longer supported. Return True if
+        1. user app version is less than minimum supported version
+        2. user app version >= minimum supported version and user app version < next minimum supported version and
+        request timestamp > upgrade deadline
+        """
+        user_app_version = self.parsed_version(user_app_version)
+        if (user_app_version < self.parsed_version(self.min_supported_version) or (
+                user_app_version < self.parsed_version(self.next_min_supported_version) and
+                datetime.now() > self.update_required_date.replace(tzinfo=None)
+        )):
+            return True
+        else:
+            return False
+
+    def is_new_version_available(self, user_app_version):
+        """ True if user is using older version of app """
+        return True if self.parsed_version(user_app_version) < self.parsed_version(self.latest_version) else False
+
+    def is_deprecated_version(self, user_app_version):
+        """
+        This is the case when there exists a higher version (than user app user) that would contain critical/major
+        application changes and would be bound with a deadline to upgrade.
+        """
+        if self.parsed_version(user_app_version) <= self.parsed_version(self.next_min_supported_version):
```

Comment:
@nasthagiri Consider we have;
1. last stable version (vA) 
1. a major change in a version and a deadline attached to it (vB) and deadline feb29.
2. a new latest version with minor changes (without any deadline and an option for user to update); ( vC released after feb29 obviously).
`min_supported_version = vA`
'deadline = feb29'
'latest_version = vC'

Now, lets consider we have two users, U1 at version vA and U2 at vB and current date feb25
Case 1. for U1; current date < deadline; we will notify user to upgrade app before deadline
Case 2. for U2; current date < deadline; we want to only notify that a new version is available.

if we do not keep track of `next_min_supported_version` along with its deadline, i wonder; we would not be able to differentiate between the two cases.